x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/textembedding/CustomElandInternalServiceSettings.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ *
+ * this file was contributed to by a generative AI
+ */
+
+package org.elasticsearch.xpack.inference.services.textembedding;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.TransportVersions.ML_TEXT_EMBEDDING_INFERENCE_SERVICE_ADDED;
+
+public class CustomElandInternalServiceSettings extends TextEmbeddingInternalServiceSettings {
+
+    public static final String NAME = "custom_eland_model_internal_service_settings";
+
+    public CustomElandInternalServiceSettings(int numAllocations, int numThreads, String modelId) {
+        super(numAllocations, numThreads, modelId);
+    }
+
+    /**
+     * Parse the CustomElandServiceSettings from map and validate the setting values.
+     *
+     * This method does not verify the model variant
+     *
+     * If required setting are missing or the values are invalid an
+     * {@link ValidationException} is thrown.
+     *
+     * @param map Source map containing the config
+     * @return The {@code CustomElandServiceSettings} builder
+     */
+    public static Builder fromMap(Map<String, Object> map) {
+
+        ValidationException validationException = new ValidationException();
+        Integer numAllocations = ServiceUtils.removeAsType(map, NUM_ALLOCATIONS, Integer.class);
+        Integer numThreads = ServiceUtils.removeAsType(map, NUM_THREADS, Integer.class);
+
+        validateParameters(numAllocations, validationException, numThreads);
+
+        String modelId = ServiceUtils.removeAsType(map, MODEL_ID, String.class);
```

Comment:
Is `modelId` required? We have a few utility functions that check for empty strings that might be helpful if it is required or optional: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/ServiceUtils.java#L184